src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs

```diff
@@ -0,0 +1,1497 @@
+            else
+            {
+                _launchTypes = LaunchType.GetBuiltInLaunchTypes().ToList<LaunchType>();
+            }
+
+            OnPropertyChanged(nameof(LaunchTypes));
+
+            // The selected launch type has to be tweaked for DotNet since in dotnet we don't want to support commands and yet user might have some commands 
+            if (!IsProfileSelected)
+            {
+                SelectedLaunchType = null;
+            }
+            else
+            {
+                var selKind = SelectedDebugProfile.Kind;
+                if ((selKind == ProfileKind.BuiltInCommand || selKind == ProfileKind.CustomizedCommand))
+                {
+                    selKind = ProfileKind.Project;
+                }
+                SelectedLaunchType = LaunchType.GetAllLaunchTypes().Where(lt => lt.Kind == selKind).SingleOrDefault();
+            }
+        }
+        public virtual Task<bool> IsWebProjectAsync()
+        {
+            if (UnconfiguredProject.Capabilities != null && UnconfiguredProject.Capabilities.Contains("DotNetCoreWeb"))
+            {
+                IsWebProject = true;
+            }
+            return System.Threading.Tasks.Task.FromResult(IsWebProject);
+        }
+
+        
+        private void EnvironmentVariables_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+        {
+            // cause the property page to be dirtied when a row is added or removed
+            OnPropertyChanged("EnvironmentVariables_Contents");
+        }
+
+        private void DebugPageViewModel_EnvironmentVariables_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            // cause the property page to be dirtied when a cell is updated
+            OnPropertyChanged("EnvironmentVariables_Contents");
+        }
+
+        /// <summary>
+        /// Overridden to do cleanup
+        /// </summary>
+        public override void ViewModelDetached()
+        {
+            if (_debugProfileProviderLink != null)
+            {
+                _debugProfileProviderLink.Dispose();
+                _debugProfileProviderLink = null;
+            }
+        }
+
+        [ExcludeFromCodeCoverage]
+        protected virtual IEnumerable<string> GetAvailableVersions()
+        {
+            return null;
+        }
+
+        [ExcludeFromCodeCoverage]
+        protected virtual ILaunchSettingsProvider GetDebugProfileProvider()
+        {
+            return UnconfiguredProject.Services.ExportProvider.GetExportedValue<ILaunchSettingsProvider>();
+        }
+
+        [ExcludeFromCodeCoverage]
+        protected virtual ISSLPortProvider GetSSLPortProvider()
+        {
+            try
+            {
+                return UnconfiguredProject.Services.ExportProvider.GetExportedValue<ISSLPortProvider>();
+            }
+            catch
+            {
+            }
+            return null;
+            
+        }
+
+        public class LaunchType
```

Comment:
>LaunchType [](start = 21, length = 10)

This will need to be completely reworked in the future to allow the web parts to be extensible.